borg/archiver.py

```diff
@@ -761,6 +761,19 @@ def build_parser(self, args=None, prog=None):
         See the output of the "borg help patterns" command for more help on exclude patterns.
         """)
 
+        class ToggleAction(argparse.Action):
+            """argparse action to handle "toggle" flags easily
+
+            toggle flags are in the form of ``--foo``, ``--no-foo``.
+
+            the ``--no-foo`` argument still needs to be passed to the
+            ``add_argument()`` call, but it simplifies the ``--no``
+            detection.
+            """
+            def __call__(self, parser, ns, values, option):
+                """set the given flag to true unless ``--no`` is passed"""
+                setattr(ns, self.dest, option.startswith('--no-'))
+
```

Comment:
I agree it's a wild mix and messy there, it could be split into a helpers package and some modules in there.
I also thought about splitting off the arg parser setup into a separate module, but this has to be evaluated - guess one would need quite some mutual imports then...